State.java

```diff
@@ -6,18 +6,46 @@ public State(String name, String abb) {
         this.abb = abb;
     }
 
-    public boolean equals(String input) {
-        if(input.equals(abb)) {
-            return true;
-        }
-        else{
-            return false;
-        }
-    }
+//    public boolean equals(String input) {
+//        if(input.equals(abb)) {
+//            return true;
+//        }
+//        else{
+//            return false;
+//        }
+//    }
     public String abbreviation() {
         return abb;
     }
     public String toString() {
         return name + " " + abb;
     }
+
+    private int toInteger(char letter)
+    {
+        return (int) letter - 97;
+    }
+
+    @Override
+    public int hashCode()
+    {
+        char firstLetter = abb.charAt( 0 );
+        char secondLetter = abb.charAt( 1 );
+        return this.toInteger( firstLetter ) * 26 + this.toInteger( secondLetter );
+    }
+
+    @Override
+    public boolean equals( Object o)
+    {
+        if ( this == o )
+        {
+            return true;
+        }
+        if ( o == null || getClass() != o.getClass() )
+        {
+            return false;
+        }
+        State state = (State) o;
+        return hashCode() == o.hashCode();
+    }
```

Comment:
here, we check a few conditions:

- if `this` object and `o` object are the same object, in other words, if they are the same object in memory, then they are equal
- if `o` is null, or if the `o`'s class is not `State` then return false
- Otherwise, do the binary hash per the directions sheet